terraformer_permissionsboundary.tf

```diff
@@ -0,0 +1,291 @@
+      aws_security_group.s3_endpoint_client.arn,
+      aws_security_group.scanner.arn,
+      aws_security_group.smb_client.arn,
+      aws_security_group.ssm_agent_endpoint_client.arn,
+      aws_security_group.ssm_endpoint_client.arn,
+      aws_security_group.sts_endpoint_client.arn,
+      aws_security_group.teamserver.arn,
+      aws_security_group.windows.arn,
+    ]
+    sid = "AllowLaunchingOpsInstancesAndAddingRemovingSGsFromInstances"
+  }
+
+  # Allow Terraformer instances to create new security groups and routing tables
+  # and manage VPC peering connections in the assessment VPC.  This explicit
+  # "allow" is necessary because the VPC is tagged with the "Team" tag.
+  statement {
+    actions = [
+      "ec2:AcceptVpcPeeringConnection",
+      "ec2:CreateRouteTable",
+      "ec2:CreateSecurityGroup",
+      "ec2:CreateVpcPeeringConnection",
+      "ec2:DeleteVpcPeeringConnection",
+    ]
+    resources = [
+      aws_vpc.assessment.arn,
+    ]
+    sid = "AllowCreatingSGsAndRoutingTablesInAssessmentVPC"
+  }
+
+  # Allow Terraformer instances to create, modify, and delete network
+  # ACLs for the operations subnet.  This explicit "allow" is necessary
+  # because the Operations NACL is tagged with the "Team" tag.
+  statement {
+    actions = [
+      "ec2:CreateNetworkAclEntry",
+      "ec2:DeleteNetworkAclEntry",
+      "ec2:ReplaceNetworkAclEntry",
+    ]
+    resources = [
+      aws_network_acl.operations.arn,
+    ]
+    sid = "AllowManagingOperationsNACL"
+  }
+
+  # Allow Terraformer instances to disassociate the default operations and
+  # private routing tables and associate additional routing tables with the
+  # first private subnet.  This is needed so that custom routing tables can
+  # be used.  This explicit "allow" is necessary because the resources below
+  # are tagged with the "Team" tag.
+  statement {
+    actions = [
+      "ec2:AssociateRouteTable",
+      "ec2:DisassociateRouteTable",
+    ]
+    resources = [
+      aws_default_route_table.operations.arn,
+      aws_route_table.private_route_table.arn,
+      aws_subnet.private[var.private_subnet_cidr_blocks[0]].arn,
+    ]
+    sid = "AllowAssociatingDisassociatingRouteTables"
+  }
+
+  # Allow Terraformer instances to modify the S3 VPC gateway endpoint.  This
+  # is needed so that the endpoint can be added to a new, custom route table.
+  # This explicit "allow" is necessary because the S3 endpoint is tagged with
+  # the "Team" tag.
+  statement {
+    actions = [
+      "ec2:ModifyVpcEndpoint",
+    ]
+    resources = [
+      aws_vpc_endpoint.s3.arn,
+    ]
+    sid = "AllowModifyingS3Endpoint"
+  }
+
+  # Don't allow Terraformer instances to touch the CloudFormation foo
+  # put in place by Control Tower.  This is not covered by the earlier
+  # statement allowing full access to resources that are not tagged as
+  # belonging to the team that deploys this root module, since CloudFormation
+  # resources do not accept tags.
```

Comment:
Maybe I'm just having a time while reading this comment, but it starts by talking about what's being denied and then says we're doing it because it's not covered by an allow?

Reply:
Take a look below at my attempt to make this read better:

```
  # Don't allow Terraformer instances to touch the CloudFormation foo put in
  # place by Control Tower.  CloudFormation resources do not accept tags and
  # thus cannot be tagged with a team tag, which is why this statement is needed
  # (an earlier statement prevents modification of any resources tagged by the
  # team that deploys this root module).
```

I'm open to suggestions if this doesn't work for you.

Reply:
That works for me. 👍

Reply:
Cool, please see https://github.com/cisagov/cool-assessment-terraform/pull/215/commits/7adcc68974453e1d9de2f9fe7a180c723d27341e.